### ISSUE TITLE
Inline reusable block

### ISSUE THREAD

**Issue by strarsis** [WordPress/gutenberg#19379]
**Is your feature request related to a problem? Please describe.**
Sometimes only a piece of text should be reused in Gutenberg, 
or some inline content, e.g. text with inline images.
With the current reusable Gutenberg block this isn't possible, 
as they work only with whole blocks.

**Describe the solution you'd like**

- Extend Gutenberg reusable blocks to have a block/inline mode
- Extend base block to have block/inline mode
- Add extra inline Gutenberg reusable block collection

**Describe alternatives you've considered**
Currently I use a [Reusable Text Blocks](https://wordpress.org/plugins/reusable-text-blocks/) plugin and shortcodes for reusing text. 
But the new Gutenberg editor is an opportunity to get rid of all other workarounds like shortcodes.

**Comment by youknowriad** [comment]
Hi There!

Gutenberg already support custom inline formats which can be used as a replacement for inline shortcodes.

**Comment by strarsis** [comment]
@youknowriad: So let's say I want to re-use a piece of text 10 times on 10 different pages. How can I create a reusable Gutenberg block for this piece of text and then place that reusable block inline into a e.g. paragraph? I don't think this is possible.

**Comment by youknowriad** [comment]
Yes, that's not possible currently. One could use the Format API to store these inline fragments in a custom CPT or something but it's quite a bit of work and if it were to be implemented in Gutenberg, it will probably be done the same way.

**Comment by tomaswallentinus** [comment]
The reusable block is awesome, I used a plugin for that before, but to use it inside a p or li would be very useful.

**Comment by Mamaduka** [comment]
I think we can close this in favor of Bits/Shortblocks API - #73929. Here's a similar request and resolution https://github.com/WordPress/gutenberg/issues/10235#issuecomment-2204179906.
